### PR TITLE
Token authorization

### PR DESCRIPTION
**Pull request by AJurczyk**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -49,6 +49,19 @@
             <version>1.18.10</version>
         </dependency>
 
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-starter-security</artifactId>
+            <version>2.2.2.RELEASE</version>
```
Comment (antonina18) [inline pom.xml]: please keep versions in properties.

File: pom.xml [hunk 2]
```diff
@@ -49,6 +49,19 @@
...
+        </dependency>
+
+        <dependency>
+            <groupId>io.jsonwebtoken</groupId>
+            <artifactId>jjwt</artifactId>
+            <version>0.9.1</version>
+        </dependency>
+
+
```
Comment (antonina18) [inline pom.xml]: empty line

File: src/main/java/pl/pongut/shop/Main.java [hunk 3]
```diff
@@ -12,4 +15,8 @@ public static void main(String[] args) {
         app.run(args);
     }
 
+    @Bean
```
Comment (antonina18) [inline src/main/java/pl/pongut/shop/Main.java]: Why it's here? Main it's not a good place to create bean.

File: src/main/java/pl/pongut/shop/config/DatabaseInitializer.java [hunk 4]
```diff
@@ -15,19 +18,35 @@
...
-                new Product(3, "Pokarm dla chomików", 7.50f)
-        ));
+        addProducts();
+        addUsers();
 
         log.info("Database initialized.");
     }
+
+    private void addUsers() {
+//        String encodedPassword = passwordEncoder.encode("kotu123");
```
Comment (antonina18) [inline src/main/java/pl/pongut/shop/config/DatabaseInitializer.java]: You shouldn't commit commented line

File: src/main/java/pl/pongut/shop/security/JwtAuthorizationFilter.java [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+        this.jwtConfig = jwtConfig;
+    }
+
+    @Override
+    protected void doFilterInternal(HttpServletRequest request,
+                                    HttpServletResponse response,
+                                    FilterChain chain) throws IOException, ServletException {
+
+        String header = request.getHeader(jwtConfig.getTokenHeader());
+        if (header == null || !header.startsWith(jwtConfig.getTokenPrefix())) {
```
Comment (antonina18) [inline src/main/java/pl/pongut/shop/security/JwtAuthorizationFilter.java]: It's good practice to make it one line method, where name means something
!header.startsWith(jwtConfig.getTokenPrefix()) < can be replaced "hasTokenPrefix" for example

File: src/main/java/pl/pongut/shop/security/WebSecurityConfig.java [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+    public WebSecurityConfig(JwtConfig jwtConfig, UserService userService, PasswordEncoder passwordEncoder) {
+        this.jwtConfig = jwtConfig;
+        this.userDetailsService = userService;
+        this.passwordEncoder = passwordEncoder;
+    }
+
+    @Override
+    protected void configure(HttpSecurity http) throws Exception {
+        http.cors().and().csrf().disable().authorizeRequests()
+                .antMatchers("/api/public", "/login").permitAll()
```
Comment (antonina18) [inline src/main/java/pl/pongut/shop/security/WebSecurityConfig.java]: please make these two strings constant

File: src/main/java/pl/pongut/shop/service/UserService.java [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+    public UserService(UserRepository userRepository) {
+        this.userRepository = userRepository;
+    }
+
+    private UserRepository userRepository;
+
+    @Override
+    public UserDetails loadUserByUsername(String username) throws UsernameNotFoundException {
+        return userRepository.findByUsername(username)
+                .map(u -> new org.springframework.security.core.userdetails.User(
```
Comment (antonina18) [inline src/main/java/pl/pongut/shop/service/UserService.java]: org.springframework.security.core.userdetails. why it's here?

File: src/main/java/pl/pongut/shop/service/UserService.java [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+                        u.getUsername(),
+                        u.getPassword(),
+                        new ArrayList<>()))
+                .orElseThrow(() -> new UsernameNotFoundException("User not found with username: " + username));
```
Comment (antonina18) [inline src/main/java/pl/pongut/shop/service/UserService.java]: User not found with username: > Username x not found?

**Review by antonina18 — CHANGES_REQUESTED** [review]

File: pom.xml (hunk 1 above)
Comment (ajurczyk1) [reply pom.xml]: done

**Review by antonina18 — APPROVED** [review]

**Pull request merged**